### PR TITLE
Add HTTPS support

### PR DESCRIPTION
**Pull request by Atte**
This pull request adds optional support for HTTPS URLs.

The public API of the library remains unchanged, and unless the `native-tls` feature is enabled the internals work more or less as before as well.

### CODE REVIEW

**Comment by viniciusgerevini** [comment]
Hello @Atte. This looks awesome. I appreciate your contribution.

I haven't being able to test it just yet, however I've noticed it broke one test related to closing the stream. I will add a comment on the line were I think the issue is.

For some reason Travis is not showing test results in this repo anymore, but here is the link in case you want to check it: https://travis-ci.org/viniciusgerevini/sse-client/builds/637972833

Btw, it is really cool you were able to make it backwards compatible 👍 .

File: src/network.rs [hunk 1]
```diff
@@ -157,20 +164,29 @@ fn listen_stream(
     });
 }
 
-fn connect_event_stream(url: &Url, stream: &StreamWrapper, last_event_id: &LastIdWrapper) -> Result<TcpStream, Error> {
+fn connect_event_stream(url: &Url, last_event_id: &LastIdWrapper) -> Result<InnerStream, Error> {
     let connection_stream = event_stream_handshake(url, last_event_id)?;
-    let mut stream = stream.lock().unwrap();
-    *stream = Some(connection_stream.try_clone().unwrap());
```
Comment (viniciusgerevini) [inline src/network.rs:163]: I believe this block is important, as the `close` method uses this reference to close the connection. If you are able to add it back it should fix the issue.

**Comment by Atte** [comment]
Whoops! Don't know how I managed to miss a failing test...

A TlsStream can't be cloned, but since the only thing the TcpStream is externally used for is closing, I don't think a handle to the full TlsStream is necessary to store in the struct. Not trying to keep the full handle in the struct allowed reverting a bunch of the changes, and removing Mutexes from MaybeTlsStream.

**Comment by viniciusgerevini** [comment]
Great! 👍 
Thanks

**Pull request merged**